lib/ansible/modules/net_tools/haproxy.py

```diff
@@ -240,6 +247,23 @@ def execute(self, cmd, timeout=200, capture_output=True):
         self.client.close()
         return result
 
+    def openSocket(self):
+        """
+        Opens a socket to HAProxy's socket, either as a UNIX socket or as
+        a TCP socket, depending on the format of the configuration option.
+        """
+        if self.socket.startswith('tcp:'):
+            match = re.search('^tcp://(.+):(\d+)$', self.socket)
```

Comment:
@phjardas 
Are we restricting to only use in `tcp://HOST:PORT`  format ?, still `tcp:HOST:PORT` is valid, `socat` allows you to read it from.
Can you please fix the regex match so that it allows to pick the HOST and PORT for admin socket even if its in `tcp://host:port` or tcp:host:port` format.